src/sentry/static/sentry/app/views/settings/projectDataPrivacy/projectDataPrivacyRulesForm.tsx

```diff
@@ -0,0 +1,202 @@
+  customRegularExpression?: string;
+};
+
+type Props = {
+  onDelete?: () => void;
+  onChange: (rule: Rule) => void;
+} & Rule;
+
+type State = {
+  errors: {
+    [key: string]: string;
+  };
+};
+class ProjectDataPrivacyRulesForm extends React.PureComponent<Props, State> {
+  state: State = {
+    errors: {},
+  };
+
+  handleChange = <T extends keyof Omit<Rule, 'id'>>(stateProperty: T, value: Rule[T]) => {
+    this.props.onChange({
+      ...omit(this.props, 'Onchange', 'onDelete'),
+      [stateProperty]: value,
+    });
+  };
+
+  handleOnDelete = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    const {onDelete} = this.props;
+    onDelete && onDelete();
+  };
+
+  handleValidation = <T extends keyof Omit<Rule, 'id'>>(field: T) => () => {
+    const errors = {...this.state.errors};
+    const isFieldValueEmpty = !this.props[field];
+    const fieldErrorAlreadyExist = errors[field];
+
+    if (isFieldValueEmpty && fieldErrorAlreadyExist) {
+      return;
+    }
+
+    if (isFieldValueEmpty && !fieldErrorAlreadyExist) {
+      errors[field] = t('Field Required');
+    }
+
+    if (!isFieldValueEmpty && fieldErrorAlreadyExist) {
+      delete errors[field];
+    }
+
+    this.setState({
+      errors,
+    });
+  };
+
+  render() {
+    const {onDelete, action, from, customRegularExpression, data} = this.props;
+    const {errors} = this.state;
+    const displayCustomRegularExpressionField =
+      data === DATA_TYPE.CUSTOM_REGULAR_EXPRESSION;
+    return (
+      <Wrapper hasError={Object.keys(errors).length > 0}>
+        <WrapperFields displaySecondRow={displayCustomRegularExpressionField}>
+          <StyledSelectControl
+            placeholder={t('Select an action')}
+            name="action"
+            options={Object.entries(ACTION_TYPE).map(([key, value]) => ({
+              label: getActionTypeSelectorFieldLabel(ACTION_TYPE[key]),
+              value,
+            }))}
+            value={action}
+            onChange={({value}) => this.handleChange('action', value)}
+            openOnFocus
+            required
+          />
+
+          <StyledSelectControl
+            placeholder={t('Select data')}
+            name="data"
+            options={Object.entries(DATA_TYPE).map(([key, value]) => ({
+              label: getDataSelectorFieldLabel(DATA_TYPE[key]),
+              value,
+            }))}
+            value={data}
+            onChange={({value}) => this.handleChange('data', value)}
+            openOnFocus
+            required
+          />
+
+          <From>{t('from')}</From>
+          <StyledTextField
+            name="from"
+            placeholder={t('ex. strings, numbers, custom')}
+            onChange={(value: string) => {
+              this.handleChange('from', value);
+            }}
+            value={from}
+            inputStyle={{
+              height: '100%',
+            }}
+            onBlur={this.handleValidation('from')}
+            error={errors.from}
```

Comment:
hmm never mind I guess we can't use input control because of this, can we try using the TextField in `app/views/settings/components/forms`? they should be compatible, but if it doesn't work, we can just leave this as is.